### ISSUE TITLE
find in files no longer works 

### ISSUE THREAD

**Issue by flottsam** [rstudio/rstudio#13216]
### System details

    RStudio Edition : <-- Desktop -->
    RStudio Version :  2023.03.1
    OS Version      :  Windows 10 Pro 19045.2965
    R Version       :  4.1.1

### Steps to reproduce the problem
'find in files' (ctrl + shif + f) to search for a function or object no longer works. 

### Describe the problem in detail
For example, searching for something basic like 'library' doe s not return anything. 
### Describe the behavior you expected
Find in files should return every instance of 'libray' within the directory being searched. super useful feature.. 

- [x ] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [ x] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [ ] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [ x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by kevinushey** [comment]
Can you please provide more details? Are you working within an RStudio project, or no project? Does that project use version control? Have you selected any other Find in Files options (e.g. "Whole Word")?

**Comment by flottsam** [comment]
Yes, using version control with git and R Studio projects. I've updated to the latest version of git (2.41.0.windows.1), but still having the issue (only with the projects that have version control). I've tried every Find in File option, nothing seems to work.

**Comment by kevinushey** [comment]
Can you please run:

```
Sys.setenv(RSTUDIO_GREP_DEBUG = 1)
```

and run your Find in Files query once more? Then, in the RStudio console, please copy and paste the output you see.

**Comment by kevinushey** [comment]
This does feel like a git bug to me. 😞

The code in git suggests that this is caused when trying to use git with FAT32 volumes:

https://github.com/git/git/blob/fe86abd7511a9a6862d5706c6fa1d9b57a63ba09/compat/mingw.c#L2755-L2759

but I'm not sure what the right way forward is, if git isn't allowing you to change those options.

**Comment by flottsam** [comment]
hmmm ok well thanks for the help.

**Comment by timtmok** [comment]
@flottsam can you confirm if you're using FAT32 for your storage?

**Comment by flottsam** [comment]
It's exfat..

**Comment by kevinushey** [comment]
Why not NTFS?

**Comment by kevinushey** [comment]
macOS has supported NTFS filesystems for some time now.

**Comment by flottsam** [comment]
OK, I didn't realize. Maybe I'll reformat the disks and try that.

**Comment by statnmap** [comment]
I faced the same problem today on Windows, although both RStudio Desktop and git are up-to-date.  
Indeed, my laptop has dual boot Windows / Linux, and I share my git repository between both sides.  

- 🛑 The error message also said to use `git config --global --add safe.directory 'D:/my_dir/my_git_project'` but this did not fix the problem.  
- 🛑   I tried `git config --global --add safe.directory *` in the RStudio Terminal as I understand the risks behind this, but this did not fix the problem either.  
- 🆗 Finally, I opened a Powershell terminal as Administrator, and run `git config --system --add safe.directory *`. 
Now the 'Find in Files' works again ! 🎉
